### PR TITLE
Use transfinite interpolation for codim 1 refinement

### PR DESCRIPTION
**Pull request by starki0815**
This patch changes the scheme used to determine middle vertices upon mesh refinement for Triangulation<2, 3> to transfinite interpolation. This makes the mesh refinement behavior consistent with what happens for Triangulation<3, 3> (see the test included). The latter is helpful if a boundary/interface mesh is extracted from a domain mesh and both need to be refined in the same way. Also, it creates nicer codim 1 meshes in some cases; see e.g. the test codim_one/extract_boundary_mesh_12 (left: before change, right after change):
comparison_extract_boundary_mesh_12

### CODE REVIEW

File: doc/news/changes/minor/20200504SebastianStark [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Changed: Compute middle vertices during mesh refinement of Triangulation<2,3> using transfinite interpolation to be consistent with what happens to quads for Triangulation<3,3>
```
Comment (kronbichler) [inline doc/news/changes/minor/20200504SebastianStark:1]: Please break the line at around 80 characters.

Comment (starki0815) [reply doc/news/changes/minor/20200504SebastianStark]: Thanks, well noted.

**Comment by masterleinad** [comment]
Expectedly, there are a number of tests failing. Can you try to compile `deal.II` with `OpenCASCADE` and check at least the `opencascade`, `manifold`, `grid` and `codim_one` tests?

```
255 - opencascade/circle_normal_projection.debug (Failed)
256 - opencascade/interpolation_curve_boundary.debug (Failed)
261 - opencascade/interpolation_curve_boundary_02.debug (Failed)
270 - opencascade/create_tria_01.debug (Failed)
275 - opencascade/stl_refinement.debug (Failed)
711 - manifold/polar_manifold_04.debug (Failed)
735 - manifold/manifold_id_07.debug (Failed)
739 - manifold/manifold_id_06.debug (Failed)
778 - manifold/manifold_id_05.debug (Failed)
3461 - grid/grid_generator_hyper_sphere.debug (Failed)
3491 - grid/grid_out_gnuplot_01.debug (Failed)
3633 - grid/grid_out_gnuplot_02.debug (Failed)
3680 - codim_one/grid_refinement.debug (Failed)
3692 - codim_one/extract_boundary_mesh_13.debug (Failed)
3714 - codim_one/extract_boundary_mesh_11.debug (Failed)
3720 - codim_one/mapping_fe_field_01.debug (Failed)
3731 - codim_one/extract_boundary_mesh_12.debug (Failed)
3732 - codim_one/boundary_indicator_01.debug (Failed)
3737 - codim_one/transform_01.debug (Failed)
```

**Comment by starki0815** [comment]
@masterleinad 

> Expectedly, there are a number of tests failing. Can you try to compile `deal.II` with `OpenCASCADE` and check at least the `opencascade`, `manifold`, `grid` and `codim_one` tests?

I've fixed the tests.

**Comment by masterleinad** [comment]
Thanks!

**Pull request merged**